src/base/DeltaResolver.sol

```diff
@@ -82,16 +82,14 @@ abstract contract DeltaResolver is ImmutableState {
     /// @notice Calculates the amount for a take action
     function _mapTakeAmount(uint256 amount, Currency currency) internal view returns (uint256) {
         if (amount == Constants.OPEN_DELTA) {
-            return _getFullCredit(currency).toUint128();
+            return _getFullCredit(currency);
         }
         return amount;
     }
 
     /// @notice Calculates the amount for a swap action
     function _mapInputAmount(uint128 amount, Currency currency) internal view returns (uint128) {
-        if (amount == Constants.CONTRACT_BALANCE) {
-            return currency.balanceOfSelf().toUint128();
-        } else if (amount == Constants.OPEN_DELTA) {
+        if (amount == Constants.OPEN_DELTA) {
             return _getFullCredit(currency).toUint128();
         }
         return amount;
```

Comment:
this function is now exactly the same as `_mapTakeAmount()` (minus casting)

1) why did we get rid of `amount == Constants.CONTRACT_BALANCE`

2) can we natspec why `_getFullCredit` is used for fetching input amounts? (i'm assuming its for multihop -- the inputs of an intermediate swap is the output (credit) of the previous step)

Reply:
> why did we get rid of amount == Constants.CONTRACT_BALANCE

I just cant see a time it will be used. The entire point in this function is for 1 usecase:
- USDC traded on v2 for UNI. UNI is left in the router
- Now we want to trade UNI for PEPE on v4 how do we do that?

Previously i thought we would do `swap with input Constants.CONTRACT_BALANCE`... and then do `settle OPEN_DELTA`. But Ive realised we're duplicating logic because you can do `settle CONTRACT_BALANCE` and then `swap OPEN_DELTA`.

Maybe its worth having both? idk just feels like a lot of options